File: Lecture-13/116. Populating Next Right Pointers in Each Node.cpp
class Solution {
public:
Node* connect(Node* root) {
    
    if(root == NULL) return NULL;
     
    if(root->left != NULL) root->left->next = root->right;
   
    if(root->right != NULL && root->next != NULL) root->right->next = root->next->left;
    
    connect(root->left);
    connect(root->right);
    return root;
   }
};

File: Lecture-13/116. Populating Next Right Pointers in Each Node.java
class Solution {
    public Node connect(Node root) {
        Node level=root;
        while(level!=null && level.left!=null){
            Node temp=level;
            while(true){
                temp.left.next=temp.right;
                if(temp.next==null) break;
                temp.right.next=temp.next.left;
                temp=temp.next;
            }
            level=level.left;
        }
        return root;
    }
}
